### PR TITLE
Documenting couple overriding env variables

### PR DESCRIPTION
**Pull request by bdematte**

### CODE REVIEW

**Comment by bdematte** [comment]
@2uasimojo here is the dedicated PR for the README update integrating your other comments

File: README.md [hunk 1]
```diff
@@ -182,3 +182,18 @@ In your fork of this repository (not a consuming repository):
...
       be useful for `update`s.
+
+## Testing
+
+To test your changes, you can use the `BOILERPLATE_GIT_REPO` environment 
+variable and set it to your local clone in order to override the version of 
+boilerplate used (Example: `export BOILERPLATE_GIT_REPO=~/git/boilerplate`). 
+
+
+Default `update` behaviour consists in cloning the git repo, so ensure you have
```
Comment (2uasimojo) [inline README.md:193]:
```suggestion
Default `update` behaviour consists of cloning the git repo, so ensure you have 
```

File: README.md [hunk 2]
```diff
@@ -182,3 +182,18 @@ In your fork of this repository (not a consuming repository):
...
+## Testing
+
+To test your changes, you can use the `BOILERPLATE_GIT_REPO` environment 
+variable and set it to your local clone in order to override the version of 
+boilerplate used (Example: `export BOILERPLATE_GIT_REPO=~/git/boilerplate`). 
+
+
+Default `update` behaviour consists in cloning the git repo, so ensure you have 
+your changes locally committed for your testing. 
+Alternativaly, you can use `BOILERPLATE_GIT_CLONE` variable to override the base
```
Comment (2uasimojo) [inline README.md:195]:
```suggestion
Alternatively, you can use the `BOILERPLATE_GIT_CLONE` variable to override the base 
```

File: README.md [hunk 3]
```diff
@@ -182,3 +182,18 @@ In your fork of this repository (not a consuming repository):
...
+To test your changes, you can use the `BOILERPLATE_GIT_REPO` environment 
+variable and set it to your local clone in order to override the version of 
+boilerplate used (Example: `export BOILERPLATE_GIT_REPO=~/git/boilerplate`). 
+
+
+Default `update` behaviour consists in cloning the git repo, so ensure you have 
+your changes locally committed for your testing. 
+Alternativaly, you can use `BOILERPLATE_GIT_CLONE` variable to override the base 
+command used for cloning the project. Example of usecases :
+- Add some flags to git clone command
```
Comment (2uasimojo) [inline README.md:197]:
```suggestion
- Add some flags to the git clone command
```

**Pull request merged**